caffe2/core/tensor_impl.h

```diff
@@ -133,12 +135,17 @@ class CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
       return;
     }
     if (data_type_ != src.meta()) {
+      CAFFE_ENFORCE_WITH_CALLER(
+          src.is_contiguous(),
+          "Source Tensor must be contiguous in order to be copied.");
```

Comment:
Nit on this error message and others: there isn't a *fundamental* reason why tensors need to be contiguous to be copied; it's just an incidental fact that the implementation doesn't support it. This is not important now, but may become more important later when we start mixing strided and non-strided tensors in Caffe2 world.